### PR TITLE
reference: add cloud related tools in doc

### PR DESCRIPTION
**Pull request by dcalvin**
### What is changed, added or deleted? 

This PR added Tools in Kubernetes for pure review purpose. So it is only added in dev, which is to be re-merged in the new folder (with new TOC). 

### What is the related PR or file link(s)? 

 https://github.com/pingcap/docs-cn/pull/1657

### CODE REVIEW

File: dev/tidb-in-kubernetes/reference/tools/tools-in-kubernetes.md [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+
+{{< copyable "shell-regular" >}}
+
+```shell
+pd-ctl -d config show
+```
+
+Assume that your local port `2379` has been occupied and you want to switch to another port. Now you need to explicitly assign a PD port for `pd-ctl`:
+
+```shell
```
Comment (aylei) [inline dev/tidb-in-kubernetes/reference/tools/tools-in-kubernetes.md]: lack `copyable` macro

File: dev/tidb-in-kubernetes/reference/tools/tools-in-kubernetes.md [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+    $ tikv-ctl --ca-path ca.pem --cert-path client.pem --key-path client-key.pem --host 127.0.0.1:20160 <subcommands>
+    ```
+
+    {{< copyable "shell-regular" >}}
+
+    ```shell
+    tikv-ctl --pd 127.0.0.1:2379 compact-cluster
+    ```
+
+* **Local Mode**：In this mode, `tikv-ctl` accesses data files of TiKV and requires running TiKV instances to be stopped. To operate in the local mode,first you need to enter the [Diagnostic Mode](/tidb-in-kubernetes/troubleshoot.md#use-the-diagnostic-mode) to turn off automatic re-starting for the TiKV instance, turn off the TiKV thread, and use the `tkctl debug` command to start in the target TiKV Pod a new container that contains the `tikv-ctl` executable. The steps are as follows:
```
Comment (aylei) [inline dev/tidb-in-kubernetes/reference/tools/tools-in-kubernetes.md]:
```suggestion
* **Local Mode**：In this mode, `tikv-ctl` accesses data files of TiKV and requires running TiKV instances to be stopped. To operate in the local mode, first you need to enter the [Diagnostic Mode](/tidb-in-kubernetes/troubleshoot.md#use-the-diagnostic-mode) to turn off automatic re-starting for the TiKV instance, stop the TiKV process, and use the `tkctl debug` command to start in the target TiKV Pod a new container that contains the `tikv-ctl` executable. The steps are as follows:
```

File: dev/tidb-in-kubernetes/reference/tools/tools-in-kubernetes.md [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+
+    1. Enter the Diagnostic mode:
+
+        {{< copyable "shell-regular" >}}
+
+        ```shell
+        kubectl annotate pod <tikv-pod-name> -n <namespace> runmode=debug
+        ```
+
+    2. Turn off the TiKV thread:
```
Comment (aylei) [inline dev/tidb-in-kubernetes/reference/tools/tools-in-kubernetes.md]:
```suggestion
    2. Stop the TiKV process:
```

File: dev/tidb-in-kubernetes/reference/tools/tools-in-kubernetes.md [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+
+        {{< copyable "shell-regular" >}}
+
+        ```shell
+        tikv-ctl --db /path/to/tikv/db size -r 2
+        ```
+
+        > **Note:**
+        >
+        >   The default db path of TiKV instances in the debug container is `/proc/1/root/var/lib/tikv/db size -r 2`
```
Comment (aylei) [inline dev/tidb-in-kubernetes/reference/tools/tools-in-kubernetes.md]:
```suggestion
        >   The default db path of TiKV instances in the debug container is `/proc/1/root/var/lib/tikv/db`
```

Sorry, I've made a typo in the CN version and I'll fix it there

File: dev/tidb-in-kubernetes/reference/tools/tools-in-kubernetes.md (hunk 1 above)
Comment (dcalvin) [inline dev/tidb-in-kubernetes/reference/tools/tools-in-kubernetes.md]:
```suggestion
{{< copyable "shell-regular" >}}

```shell
```

File: dev/tidb-in-kubernetes/reference/tools/tools-in-kubernetes.md [hunk 5]
```diff
@@ -0,0 +1,241 @@
...
+```
+
+```
+NAME                    CHART VERSION   APP VERSION DESCRIPTION
+pingcap/tidb-backup     v1.0.0-rc.1                 A Helm chart for TiDB Backup or Restore
+pingcap/tidb-cluster    v1.0.0-rc.1                 A Helm chart for TiDB Cluster
+pingcap/tidb-operator   v1.0.0-rc.1                 tidb-operator Helm chart for Kubernetes
+```
+
+When a new version of chart has been released, you can use `helm repo update` to update the Repo cached locally:
```
Comment (TomShawn) [inline dev/tidb-in-kubernetes/reference/tools/tools-in-kubernetes.md]:
```suggestion
When a new version of chart has been released, use `helm repo update` to update the repository cached locally:
```

File: dev/tidb-in-kubernetes/reference/tools/tools-in-kubernetes.md [hunk 6]
```diff
@@ -0,0 +1,241 @@
...
+
+{{< copyable "shell-regular" >}}
+
+```
+helm repo update
+```
+
+Common Helm operations include `helm install`, `helm upgrade`, and `helm del`. Helm chart usually contains many configurable parameters which could be tedious to configure manually. For convenience, it is recommended you configure using a YAML file. Based on the conventions in the Helm community, we name our YAML file used for Helm configuration `values.yaml`.
```
Comment (TomShawn) [inline dev/tidb-in-kubernetes/reference/tools/tools-in-kubernetes.md]:
```suggestion
Common Helm operations include `helm install`, `helm upgrade`, and `helm del`. Helm chart usually contains many configurable parameters which could be tedious to configure manually. For convenience, it is recommended to configure using a YAML file. Based on the conventions in the Helm community, the YAML file used for Helm configuration is named `values.yaml` in this document.
```

File: dev/tidb-in-kubernetes/reference/tools/tools-in-kubernetes.md [hunk 7]
```diff
@@ -0,0 +1,241 @@
...
+
+* Upgrade (upgrade can be done by modifying the `chart-version` to upgrade to the latest chart version or the `values.yaml` file to update the configuration):
+
+    {{< copyable "shell-regular" >}}
+
+    ```shell
+    helm upgrade <release-name> <chart-name> --version=<chart-version> -f <values-file>
+    ```
+
+* If you want to delete the application deployed by Helm, use the following command:
```
Comment (TomShawn) [inline dev/tidb-in-kubernetes/reference/tools/tools-in-kubernetes.md]:
```suggestion
* To delete the application deployed by Helm, run the following command:
```

File: dev/tidb-in-kubernetes/reference/tools/tools-in-kubernetes.md [hunk 8]
```diff
@@ -0,0 +1,241 @@
...
+
+```shell
+helm del --purge <release-name>
+```
+
+For more information on Helm, refer to [Helm Documentation](https://helm.sh/docs/).
+
+## Use Terraform
+
+[Terraform](https://www.terraform.io/) is a Infrastructure as Code management tool. It enables users to define their own infrastructure in a  manifestation style, based on which execution plans are generated to create or schedule real world compute resources. TiDB in Kubernetes use Terraform to create and manage TiDB clusters on public clouds.
```
Comment (TomShawn) [inline dev/tidb-in-kubernetes/reference/tools/tools-in-kubernetes.md]:
```suggestion
[Terraform](https://www.terraform.io/) is an Infrastructure as Code management tool. It enables users to define their own infrastructure in a  manifestation style, based on which execution plans are generated to create or schedule real-world compute resources. TiDB in Kubernetes uses Terraform to create and manage TiDB clusters on public clouds.
```

File: dev/tidb-in-kubernetes/reference/tools/tools-in-kubernetes.md [hunk 9]
```diff
@@ -0,0 +1,241 @@
...
+
+After the above command is executed, you can access the PD service via `127.0.0.1:2379`, and then use the default parameters of `pd-ctl` to operate. For example:
+
+{{< copyable "shell-regular" >}}
+
+```shell
+pd-ctl -d config show
+```
+
+Assume that your local port `2379` has been occupied and you want to switch to another port. Now you need to explicitly assign a PD port for `pd-ctl`:
```
Comment (TomShawn) [inline dev/tidb-in-kubernetes/reference/tools/tools-in-kubernetes.md]:
```suggestion
Assume that your local port `2379` has been occupied and you want to switch to another port. Then you need to explicitly assign a PD port for `pd-ctl`:
```

File: dev/tidb-in-kubernetes/reference/tools/tools-in-kubernetes.md [hunk 10]
```diff
@@ -0,0 +1,241 @@
...
+
+```shell
+pd-ctl -u 127.0.0.1:<port> -d config show
+```
+
+## Use TiKV Control in Kubernetes
+
+[TiKV Control](reference/tools/tikv-control.md) is the command line tool for TiKV. When using TiKV Control for TiDB clusters in Kubernetes, be aware that different operation mode may involve different steps.
+
+* **Remote Mode**: In this mode, `tikv-ctl` accesses the TiKV service or the PD service through network, so firstly you need to establish the connection from local to the PD service and the target TiKV node using `kubectl port-forward`:
```
Comment (TomShawn) [inline dev/tidb-in-kubernetes/reference/tools/tools-in-kubernetes.md]:
```suggestion
* **Remote Mode**: In this mode, `tikv-ctl` accesses the TiKV service or the PD service through network. Therefore, you first need to establish the connection from local to the PD service and the target TiKV node using `kubectl port-forward`:
```

File: dev/tidb-in-kubernetes/reference/tools/tools-in-kubernetes.md [hunk 11]
```diff
@@ -0,0 +1,241 @@
...
+    kubectl port-forward -n <namespace> svc/<cluster-name>-pd 2379:2379 &>/tmp/portforward-pd.log
+    ```
+
+    {{< copyable "shell-regular" >}}
+
+    ```shell
+    kubectl port-forward -n <namespace> <tikv-pod-name> 20160:20160 &>/tmp/portforward-tikv.log
+    ```
+
+    After the connection is established, you can access the PD service and the TiKV node via the corresponding port in local:
```
Comment (TomShawn) [inline dev/tidb-in-kubernetes/reference/tools/tools-in-kubernetes.md]:
```suggestion
    After the connection is established, access the PD service and the TiKV node via the corresponding port in local:
```

File: dev/tidb-in-kubernetes/reference/tools/tools-in-kubernetes.md [hunk 12]
```diff
@@ -0,0 +1,241 @@
...
+
+   Confirm that the tiller pod is in the `running` state by the following command:
+
+   {{< copyable "shell-regular" >}}
+
+    ```shell
+    kubectl get po -n kube-system -l name=tiller
+    ```
+
+   If `RBAC` is not enabled for the Kubernetes cluster, then use the following command to install `tiller`:
```
Comment (TomShawn) [inline dev/tidb-in-kubernetes/reference/tools/tools-in-kubernetes.md]:
```suggestion
   If `RBAC` is not enabled for the Kubernetes cluster, use the following command to install `tiller`:
```

File: dev/tidb-in-kubernetes/reference/tools/tools-in-kubernetes.md (hunk 11 above)
Comment (dcalvin) [reply dev/tidb-in-kubernetes/reference/tools/tools-in-kubernetes.md]: This is a description to indicate a result of the previous operations - that you can do something. So I think we should keep "you can" here

File: dev/tidb-in-kubernetes/reference/tools/tools-in-kubernetes.md [hunk 13]
```diff
@@ -0,0 +1,241 @@
...
+kubectl port-forward -n <namespace> svc/<cluster-name>-pd 2379:2379 &>/tmp/portforward-pd.log
+```
+
+{{< copyable "shell-regular" >}}
+
+```shell
+kubectl port-forward -n <namespace> <tidb-pod-name> 10080:10080 &>/tmp/portforward-tidb.log
+```
+
+Then you can use the `tidb-ctl`:
```
[the remark this answers is not in the document]
Comment (dcalvin) [reply dev/tidb-in-kubernetes/reference/tools/tools-in-kubernetes.md]: ditto.

**Comment by dcalvin** [comment]
PTAL again. Thanks.

**Pull request merged**